sensor/common/sensor/central_communication_impl.go

```diff
@@ -142,6 +145,9 @@ func (s *centralCommunicationImpl) sendEvents(client central.SensorServiceClient
 	}
 
 	defer func() {
+		s.stoppedC.SignalWithError(s.stopC.Err())
+		runAll(s.stopC.Err(), onStops...)
+		s.sender.(*centralSenderImpl).waitC.Wait()
```

Comment:
After talking with @fredrb about this, I think we shouldn't close the connection or stop Sensor in the tests for now (see the [comment](https://github.com/stackrox/stackrox/pull/2202#pullrequestreview-1034123665) in #2202)

Answering your question, I don't think the cast could fail but checking it wouldn't hurt just to be on the safe side. Nevertheless, since we are not going to close or stop Sensor in these tests I'll revert this change.